src/home_robot/home_robot/utils/point_cloud_torch.py

```diff
@@ -140,38 +196,39 @@ def grid_pool_point_cloud(
     Overlays a grid, and selects one point in each grid cell (if one exists). If use_random_centers is True,
     the point selected is random within that cell. Otherwise it's whichever voxel_grid returns first.
     """
-    # Voxel grid returns a list, same length as the original, with a mapping to unique grid identifiers
-    xyz_grid_indices = voxel_grid(unbatched_xyz, voxel_size, unbatched_batch_ids)
-
-    # We wish to take one of each grid identifier, to use as our point
-    # Based on: https://stackoverflow.com/questions/72001505/how-to-get-unique-elements-and-their-firstly-appeared-indices-of-a-pytorch-tenso
-    grid_ids, xyz_to_grid_id, grid_cell_counts = torch.unique(
-        xyz_grid_indices, sorted=True, return_inverse=True, return_counts=True
-    )
-
-    # The grid ids are sorted, and the counts match that sorting. So if we sort xyz_to_grid_id and get the mapping
-    # from that operation, we can use the count maps as indices into the original xyzs, by doing cumsum
-    _, xyz_to_grid_id_sort_mapping = torch.sort(xyz_to_grid_id, stable=True)
-
-    if use_random_centers:
-        random_offsets = (
-            torch.rand(grid_cell_counts.shape[0]).to(grid_cell_counts.device)
-            * grid_cell_counts
-        ).int()
-    else:
-        random_offsets = 0
-
-    unique_grid_indices = (
-        torch.cat(
-            (
-                torch.tensor([0]).to(grid_cell_counts.device),
-                grid_cell_counts.cumsum(0)[:-1],
-            ),
-            axis=0,
-        )
-        + random_offsets
-    )
-    unique_grid_xyz_indices = xyz_to_grid_id_sort_mapping[unique_grid_indices]
-
-    # We return the indices into the original data, for consistency with fps
-    return unique_grid_xyz_indices
+    raise NotImplementedError
```

Comment:
I think that I just didn't understand what it was supposed to do, and since it was never used but required torch_geometric (which wasn't in environment.yml), I added what should be a DeprecationError.
However, we now need torch_geometric and so I renamed the function to `get_one_point_per_voxel_from_pointcloud` and restored the functionality :)
Was this a function that you added?